src/components/DictionaryCrud/DictionaryCrudPresentation.js

```diff
@@ -29,6 +29,10 @@ function DictionaryCrudPresentation({
   isCreate,
   partsOfSpeech,
 }) {
+  const [currentLinks, setCurrentLinks] = useState(
```

Comment:
I feel like we should try to avoid maintaining state related to a form field wherever possible. Also, this state management for this field is being duplicated in multiple files in this PR.
My suggestion would be to add a new `type` to the `MediaArrayField` component, like _videoLinks_ or _externalVideoLinks_ or something similar, and handle it separately by just passing the control and register and not passing in state and setState methods.

Also, we can possibly remove this as an additional tab from the uploadMedia dialog box and add an extra field in form like _wordTranslations_ which also supports multiple values.
Paging @gmcauliffe for confirmation of the placement of the field.

That would remove most of this duplication and would also make debugging and management of this field more easier as well.
Let me know and I can jump on a call to discuss.